src/transformers/pipelines/object_detection.py

```diff
@@ -108,22 +108,42 @@ def preprocess(self, image):
         image = self.load_image(image)
         target_size = torch.IntTensor([[image.height, image.width]])
         inputs = self.feature_extractor(images=[image], return_tensors="pt")
+        if self._is_layout_detection():
+            encoded_inputs = self.tokenizer(
+                text=inputs["words"],
+                boxes=inputs["boxes"],
+                truncation=True,
+                return_offsets_mapping=True,
+                return_tensors="pt",
+            )
+            encoded_inputs["image"] = inputs.pop("pixel_values")
+            encoded_inputs["target_size"] = target_size
+            return encoded_inputs
```

Comment:
I see you're using the feature extractor and tokenizer separately here. Note that you can just use the processor instead:
```suggestion
        if self._is_layout_detection():
            encoded_inputs = self.processor(image, truncation=True, return_offsets_mapping=True, return_tensors="pt")
            return encoded_inputs
```

Reply:
Because pipelines don't want to rely on the `processor`. `processor` is a concept that does not exist for most models, all the code exists to support loading `tokenizers` and `feature_extractor` automatically and magically, so using `processor` is really not worth the effort (especially since we're using a very small subset of what `processor` logic does)

Reply:
Ok, however at some point I guess we will have an `AutoProcessor` API that will load the appropriate preprocessing thing (either a `feature_extractor`, a `tokenizer` or a `processor`). For now, using the `feature_extractor` and `tokenizer` separately is fine.

Reply:
Closing it as resolved for now